runtime/common/MeasureCounts.cpp

```diff
@@ -40,25 +65,25 @@ ExecutionResult::ExecutionResult(CountsDictionary c, std::string name, double e)
 ExecutionResult::ExecutionResult(CountsDictionary c, double e)
     : counts(c), expectationValue(e) {}
 ExecutionResult::ExecutionResult(const ExecutionResult &other)
-    : counts(other.counts), expectationValue(other.expectationValue),
-      registerName(other.registerName), sequentialData(other.sequentialData) {}
+    : counts(std::move(other.counts)),
+      expectationValue(std::move(other.expectationValue)),
+      registerName(std::move(other.registerName)),
+      sequentialData(std::move(other.sequentialData)) {}
 
 ExecutionResult &ExecutionResult::operator=(const ExecutionResult &other) {
-  counts = other.counts;
-  expectationValue = other.expectationValue;
-  registerName = other.registerName;
-  sequentialData = other.sequentialData;
+  counts = std::move(other.counts);
+  expectationValue = std::move(other.expectationValue);
+  registerName = std::move(other.registerName);
+  sequentialData = std::move(other.sequentialData);
```

Comment:
Invoking `std::move() ` inside these constructors is undesirable because the input parameters (`other.*`) for these constructors are `const` and therefore need to retain ownership over their data. In this case, I think the compiler is saving you by _preventing_ the move constructor of the underlying objects from being called, which is a good thing, but it also means that the `std::move` is having no effect. Unless I'm missing something, I think these particular changes should be removed.